File: C++/Learing-C++/20. C++ OOPs/1. Class & Object/class_object.cpp
#include <iostream>
#include <string>

using namespace std;

// Define a class named 'Person'
class Person {
public:
    // Data members (attributes)
    string name;
    int age;

    // Member function (method)
    void displayInfo() {
        cout << "Name: " << name << "\n";
        cout << "Age: " << age << "\n";
    }
};

int main() {
    // Create an object of the 'Person' class
    Person person1;

    // Set values for the object's attributes
    person1.name = "John";
    person1.age = 25;

    // Call the member function to display information
    cout << "Person 1 Information:\n";
    person1.displayInfo();

    // Create another object and set values using constructor
    Person person2{"Alice", 30};

    // Display information for the second person
    cout << "\nPerson 2 Information:\n";
    person2.displayInfo();

    return 0;
}

File: Dart/Smart Dart/16. Dart OOPs/01. Class & Object/class_object.dart
// File: class_and_object_no_constructor.dart

// 1. Class
class Person {
  // 2. Properties (Attributes)
  String? name;
  int? age;

  // 3. Method
  void displayInfo() {
    print('$name, $age years old');
  }
}

void main() {
  // 4. Creating Objects without Constructor
  Person person1 = Person()
    ..name = 'Alice'
    ..age = 25;

  Person person2 = Person();
  person2.name = 'Bob';
  person2.age = 30;

  // 5. Accessing Properties and Methods
  print('Person 1:');
  person1.displayInfo();

  print('\nPerson 2:');
  person2.displayInfo();
}


/*
  Note:
  This Dart file provides a simple example of classes and objects without using a separate constructor.

  1. **Class:**
     - A class is a blueprint for creating objects. It defines the properties (attributes) and
       behavior (methods) that the objects will have.

  2. **Object:**
     - An object is an instance of a class. It represents a specific entity created based on the
       blueprint provided by the class.

*/